src/lib/draw/polygon_tool.ts

```diff
@@ -10,10 +12,22 @@ import {
   isPolygon,
   isLine,
 } from "../../maplibre_helpers";
-import { colors, circleRadius } from "../../colors.js";
+import { colors, circleRadius } from "../../colors";
+
+const source = "edit-polygon-mode";
 
 export class PolygonTool {
-  constructor(map) {
+  map: Map;
+  active: boolean;
+  eventListeners: ((f: Feature<Polygon>) => void)[];
+  points: Position[];
+  cursor: Feature<Point> | null;
+  hoverPolyon: boolean;
```

Comment:
Now that we're TS, we could get fancier here and express that hovering is either nothing | the polygon | an index point. I'll look into that later